source/src/Run.cc

```diff
@@ -78,80 +71,52 @@ namespace dqm4hep {
 
     //-------------------------------------------------------------------------------------------------
 
-    xdrstream::Status Run::stream(xdrstream::StreamingMode mode, xdrstream::IODevice *pDevice,
-        xdrstream::xdr_version_t version)
-    {
-      if( xdrstream::XDR_READ_STREAM == mode )
-      {
-        XDR_STREAM( pDevice->read<int32_t>( & m_runNumber ) )
-
-		    int64_t startTime;// = std::chrono::system_clock::to_time_t(m_startTime);
-        int64_t endTime; //= std::chrono::system_clock::to_time_t(m_endTime);
-        XDR_STREAM( pDevice->read<int64_t>( & startTime ) )
-        XDR_STREAM( pDevice->read<int64_t>( & endTime ) )
-        m_startTime = std::chrono::system_clock::from_time_t(startTime);
-        m_endTime = std::chrono::system_clock::from_time_t(endTime);
-
-        XDR_STREAM( pDevice->read( & m_detectorName ) )
-        XDR_STREAM( pDevice->read( & m_description ) )
-        XDR_STREAM( StreamingHelper::read( pDevice , m_parametersMap ))
-      }
-      else
-      {
-        XDR_STREAM( pDevice->write<int32_t>( & m_runNumber ) )
-
-		    int64_t startTime = std::chrono::system_clock::to_time_t(m_startTime);
-        int64_t endTime = std::chrono::system_clock::to_time_t(m_endTime);
-        XDR_STREAM( pDevice->write<int64_t>( & startTime ) )
-        XDR_STREAM( pDevice->write<int64_t>( & endTime ) )
-
-        XDR_STREAM( pDevice->write( & m_detectorName ) )
-        XDR_STREAM( pDevice->write( & m_description ) )
-        XDR_STREAM( StreamingHelper::write( pDevice , m_parametersMap ))
-      }
-
-      return xdrstream::XDR_SUCCESS;
-    }
-
-    //-------------------------------------------------------------------------------------------------
-
-    void Run::toJson(Json::Value &value) const
+    void Run::toJson(json &value) const
     {
-      Json::Value::Int64 startTime = std::chrono::system_clock::to_time_t(m_startTime);
-      Json::Value::Int64 endTime = std::chrono::system_clock::to_time_t(m_endTime);
-
-      value["runNumber"] = m_runNumber;
-      value["startTime"] = startTime;
-      value["endTime"] = endTime;
-      value["detector"] = m_detectorName;
-      value["description"] = m_description;
-
-      Json::Value parametersValue;
-
-      for(const auto &parameter : m_parametersMap)
-        parametersValue[parameter.first] = parameter.second;
-
-      value["parameters"] = parametersValue;
+      auto startTime = std::chrono::system_clock::to_time_t(m_startTime);
+      auto endTime = std::chrono::system_clock::to_time_t(m_endTime);
+
+      value = {
+        {"runNumber", m_runNumber},
+        {"startTime", startTime},
+        {"endTime", endTime},
+        {"detector", m_detectorName},
+        {"description", m_description},
+        {"parameters", m_parametersMap}
+      };
+      // value["runNumber"] = m_runNumber;
```

Comment:
Can probably remove the comments if the new interface is working

Reply:
ooops